cmd/ceremony/cert.go

```diff
@@ -209,7 +212,19 @@ func generateSKID(pk []byte) ([]byte, error) {
 }
 
 // makeTemplate generates the certificate template for use in x509.CreateCertificate
-func makeTemplate(randReader io.Reader, profile *certProfile, pubKey []byte, ct certType) (*x509.Certificate, error) {
+func makeTemplate(randReader io.Reader, profile *certProfile, pubKey []byte, tbcs *x509.Certificate, ct certType) (*x509.Certificate, error) {
+	// Handle "unrestricted" vs "restricted" subordinate CA profile specifics.
+	if ct == crossCert {
+		if tbcs == nil {
+			return nil, fmt.Errorf("toBeCrossSigned cert field was nil, but was required to gather EKUs for the lint cert")
+		}
+	} else {
+		// Throw this parameter away for every other ceremony type.
+		if tbcs != nil {
+			tbcs = nil
+		}
+	}
```

Comment:
This can just be simplified to:
```suggestion
	if ct == crossCert && tbcs == nil {
		return nil, fmt.Errorf("toBeCrossSigned cert field was nil, but was required to gather EKUs for the lint cert")
	}
```
We can combine the first two conditionals into a single conjunction, and we can get rid of the else clause because we're simply not going to use the tbcs.